### ISSUE TITLE
Uncaught TypeError: Cannot read property 'push' of undefined

### ISSUE THREAD

**Issue by robmercier00** [queenp/punchclock#4]
[Enter steps to reproduce below:]

1. Open Atom with a blank document, and no project
2. Press Ctrl-Cmd-S

**Atom Version**: 1.0.2
**System**: Mac OS X 10.10.4
**Thrown From**: [punchclock](https://github.com/queenp/punchclock) package, v0.6.2

### Stack Trace

Uncaught TypeError: Cannot read property 'push' of undefined

```
At ~/.atom/packages/punchclock/lib/timer.coffee:112

TypeError: Cannot read property 'push' of undefined
  at Timer.module.exports.Timer.start (~/.atom/packages/punchclock/lib/timer.coffee:112:24)
  at atom-workspace.<anonymous> (~/.atom/packages/punchclock/lib/punchclock.coffee:53:75)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:241:29)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:520:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:355:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:177:20)

```

### Commands

```
     -0:00.8.0 punchclock:start (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "disabledPackages": [
      "wrap-guide",
      "welcome",
      "language-sql-mysql",
      "linter-php",
      "linter"
    ],
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ]
  },
  "punchclock": {}
}
```

### Installed Packages

```coffee
# User
grammar-token-limit, v0.1.1
minimap, v4.12.0
punchclock, v0.6.2
wakatime, v4.0.12

# Dev
No dev packages
```

**Comment by queenp** [comment]
Cheers for this. 
I updated the code recently and inadvertently introduced this bug. I'll rewrite it to not throw an error (and instead to notify the user that no project folder is open) in the meantime, but it begs a question:

Currently Punchclock recognises/registers projects based on the Top project root directory (because Atom doesn't really have any other means of recognising/tracking projects). Presumably as you were trying to start time tracking on an unspecified project, you'd be interested in tracking time on a workspace when there's no current project. I was wondering if you could share with me what sort of behaviour you'd be expecting from this situation?

**Comment by robmercier00** [comment]
Yeah, I did notice that the update fixed a different error, which I was also encountering frequently. Once that was fixed, this is really the last remaining issue I've been encountering.

In trying to streamline and use a single text editor as much as possible (I do still have to open MS Word or LibreOffice for some word processor-type things), there are 2 scenarios that I encounter frequently where tracking time would be handy:
1) A single-page script that is not part of an existing project (a non-project, single-page php script, for example). Since it's a standalone script it would be great to be able to track time spent on this kind of development without having to make it part of a project
2) Non-development time tracking; I like using Atom as a general text editor, even for things that are not development related

I also use WakaTime, and the way they're able to track non-project work is to just group all files under a generic "unknown project" moniker. The limitation there is that I'm unable to see the time taken on each file, specifically, but that hasn't proven to be especially problematic.

I really love this plugin, btw. You've done a fantastic job with it!
